src/pkcs1v15.rs

```diff
@@ -214,6 +219,162 @@ fn non_zero_random_bytes<R: RngCore + CryptoRng>(rng: &mut R, data: &mut [u8]) {
     }
 }
 
+pub struct RsaPkcs1v15SigningKey {
+    inner: RsaPrivateKey,
+    hash: Option<Hash>,
+}
+
+impl<'a> RsaPkcs1v15SigningKey {
+    pub(crate) fn key(&'a self) -> &'a RsaPrivateKey {
+        &self.inner
+    }
+
+    pub(crate) fn hash(&self) -> Option<Hash> {
+        self.hash
+    }
+}
+
+impl From<RsaPrivateKey> for RsaPkcs1v15SigningKey {
+    fn from(key: RsaPrivateKey) -> Self {
+        Self {
+            inner: key,
+            hash: None,
+        }
+    }
+}
+
+impl From<&RsaPrivateKey> for RsaPkcs1v15SigningKey {
+    fn from(key: &RsaPrivateKey) -> Self {
+        Self {
+            inner: RsaPrivateKey::from_components(
+                key.n().clone(),
+                key.e().clone(),
+                key.d().clone(),
+                key.primes().clone().to_vec(),
+            )
+            .unwrap(),
+            hash: None,
+        }
+    }
+}
+
+impl From<(RsaPrivateKey, Hash)> for RsaPkcs1v15SigningKey {
+    fn from(data: (RsaPrivateKey, Hash)) -> Self {
```

Comment:
This feels like it should probably just be:

```rust
impl SigningKey {
    pub fn new(key: RsaPrivateKey, hash: Hash) -> Self {
```

Reply:
Heh. At some point you wrote about using `From`, so I followed that suggestion blindly. I'll change this to use `::new()`.